webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -309,9 +309,21 @@ trait PromiseGenerationService {
 
     val generationContinueOnError = getGenerationContinueOnError().getOrElse(false)
 
+    val delay = getDelay() match {
+      case Full(delay) => delay
+      case eb : EmptyBox => Duration("Os")
+    }
+
     PolicyLogger.debug(s"Policy generation parallelism set to: ${maxParallelism} (change with REST API settings parameter 'rudder_generation_max_parallelism')")
     PolicyLogger.debug(s"Policy generation JS evaluation of directive parameter timeout: ${jsTimeout} s (change with REST API settings parameter 'rudder_generation_jsTimeout')")
     PolicyLogger.debug(s"Policy generation continues on NodeConfigurations evaluation: ${generationContinueOnError} (change with REST API settings parameter 'rudder_generation_continue_on_error')")
+    if (delay.toMillis == 0L) {
+      PolicyLogger.debug(s"Policy generation will now start!")
+    } else {
+      PolicyLogger.debug(s"Policy generation will now hang for ${delay.toString} before effective start")
+    }
+
+    Thread.sleep(delay.toMillis)
```

Comment:
I don't think we want it to be here. It should be out of generation duration at least, and I think from a code point of view, it should be out of the `deploy` method. Can it be just before its call?